views/LocationListView/LocationListView.tsx

```diff
@@ -29,17 +30,25 @@ export const LocationListView = () => {
     });
   };
 
-  useEffect(() => {
-    const locations = BOT_NAMES.map(name => resolveUserID(name));
-    Promise.all(locations).then(credentials =>
-      setLocationCredentials(credentials),
-    );
-  }, []);
+  const retrieveAvailableLocations = () => {
```

Comment:
It can be placed directly inside useFocusEffect

Reply:
Check the current solution